agents/unix/conf/base/conf_pci.c

```diff
@@ -2102,15 +2145,30 @@ pci_driver_set(unsigned int gid, const char *oid, const char *value,
         rc = maybe_load_driver(value);
         if (rc != 0)
             return rc;
-        for (i = 0; i < TE_ARRAY_LEN(generic_drivers) && !is_gen_driver; i++)
-            is_gen_driver = (strcmp(generic_drivers[i], value) == 0);
 
-        if (is_gen_driver)
+        /* Available on kernels >= 3.15. */
+        rc = try_override(dev, value);
+
+        if (TE_RC_GET_ERROR(rc) == TE_EOPNOTSUPP)
         {
-            rc = let_generic_driver_know_pci_device(dev, value);
-            if (rc != 0)
-                return rc;
+            for (i = 0; i < TE_ARRAY_LEN(generic_drivers) && !is_gen_driver;
+                 ++i)
```

Comment:
According to https://github.com/oktetlabs/styleguides/blob/master/cguide.md#for-statement, if the `for` header does not fit into a single line, it should be split in three. Also you're changing a post-increment to a pre-increment without any purpose.